overlord/snapstate/snapstate.go

```diff
@@ -825,6 +825,12 @@ func FinishTaskWithRestart(task *state.Task, status state.Status, rt restart.Res
 	// task belongs to is configured (system-restart-immediate) to
 	// choose whether request an immediate restart or not.
 	if rt == restart.RestartSystem {
+		snapsup, err := TaskSnapSetup(task)
+		if err != nil {
+			return err
+		}
+		task.Set("reboot-required-snap", snapsup.InstanceName())
```

Comment:
I understand this avoids circular import issues but is strange

Reply:
I guess we have some options here:
1. keep as is
2. do not add the information to the file what snap caused the reboot
3. extract TaskSnapSetup into a new go module like `overlord/snapstate/snapstatehelpers` and use that in both places
4. re-implement the small part of `TaskSnapSetup()` that is needed in restart
maybe there is more but I'm happy to peruse any of those